test/test-instance-profiling.js

```diff
@@ -55,12 +54,10 @@ test('Check heap and cpu profiling populates Profile models', function(t) {
 
       t.test('heap snapshot delete', function(tt) {
         TestServiceManager.prototype.onCtlRequest = assertCtlRequests(tt, 1, 1);
-        instance.processes({where: {id: 1}}, function(err, processes) {
+        instance.heapSnapshot(1, function(err, res) {
           tt.ifError(err);
 
-          processes[0].heapSnapshot(function(err, res) {
-            tt.ifError(err);
-
+          setImmediate(function() {
```

Comment:
Is this `setImmediate()` necessary? If so, why? Is it covering up a bug? Feel free to answer in the form of a comment in the code if appropriate.